pkg/plugins/runtime/k8s/controllers/configmap_controller.go

```diff
@@ -60,195 +60,143 @@ func (r *ConfigMapReconciler) Reconcile(req kube_ctrl.Request) (kube_ctrl.Result
 }
 
 func (r *ConfigMapReconciler) SetupWithManager(mgr kube_ctrl.Manager) error {
-	for _, addToScheme := range []func(*kube_runtime.Scheme) error{kube_core.AddToScheme, mesh_k8s.AddToScheme} {
-		if err := addToScheme(mgr.GetScheme()); err != nil {
-			return err
-		}
-	}
 	return kube_ctrl.NewControllerManagedBy(mgr).
 		For(&kube_core.ConfigMap{}).
-		Watches(&kube_source.Kind{Type: &kube_core.Service{}}, &kube_handler.EnqueueRequestsFromMapFunc{
-			ToRequests: &ServiceToConfigMapsMapper{
-				Client:          mgr.GetClient(),
-				Log:             r.Log.WithName("service-to-configmap-mapper"),
-				SystemNamespace: r.SystemNamespace,
-			},
-		}).
-		Watches(&kube_source.Kind{Type: &mesh_k8s.Dataplane{}}, &kube_handler.EnqueueRequestsFromMapFunc{
-			ToRequests: &DataplaneToMeshMapper{
-				Client:            mgr.GetClient(),
-				Log:               r.Log.WithName("dataplane-to-configmap-mapper"),
-				SystemNamespace:   r.SystemNamespace,
-				ResourceConverter: r.ResourceConverter,
-			},
-		}).
-		Watches(&kube_source.Kind{Type: &mesh_k8s.ZoneIngress{}}, &kube_handler.EnqueueRequestsFromMapFunc{
-			ToRequests: &ZoneIngressToMeshMapper{
-				Log:               r.Log.WithName("zone-ingress-to-configmap-mapper"),
-				SystemNamespace:   r.SystemNamespace,
-				ResourceConverter: r.ResourceConverter,
-			},
-		}).
-		Watches(&kube_source.Kind{Type: &mesh_k8s.VirtualOutbound{}}, &kube_handler.EnqueueRequestsFromMapFunc{
-			ToRequests: &VirtualOutboundToConfigMapsMapper{
-				Client:          mgr.GetClient(),
-				Log:             r.Log.WithName("virtualoutbound-to-configmap-mapper"),
-				SystemNamespace: r.SystemNamespace,
-			},
-		}).
-		Watches(&kube_source.Kind{Type: &mesh_k8s.ExternalService{}}, &kube_handler.EnqueueRequestsFromMapFunc{
-			ToRequests: &ExternalServiceToConfigMapsMapper{
-				Client:          mgr.GetClient(),
-				Log:             r.Log.WithName("external-service-to-configmap-mapperr"),
-				SystemNamespace: r.SystemNamespace,
-			},
-		}).
+		Watches(&kube_source.Kind{Type: &kube_core.Service{}}, kube_handler.EnqueueRequestsFromMapFunc(ServiceToConfigMapsMapper(mgr.GetClient(), r.Log, r.SystemNamespace))).
```

Comment:
All these helper functions are only used here, why not have them return a `kube_handler.EventHandler` directly? i.e. do the `EnqueueRequestsFromMapFunc` wrapping internally ...

Reply:
It's simpler for testing to not have the whole EventHandler wrapper.